### DISCUSSION TITLE
Using nnn preview-tui plugin on sshed remote server

### DISCUSSION THREAD

**Discussion by alexisthual** [kovidgoyal/kitty#6241] [Q&A] [answered]
Good evening,

First thank you for the awesome terminal!

I have struggled with the documentation a lot today but couldn't get the following use case to work:
- I have installed `kitty` on my local machine
- I have installed `nnn` on a remote machine `a`
- I am trying to use the `preview-tui` plugin from `nnn` on this remote machine but it fails with:

```
$KITTY_LISTEN_ON not set!
Please read Usage in '/root/.config/nnn/plugins/preview-tui'.
```

Here is how I proceeded:
1. In a local terminal, I ran `kitty -o allow_remote_control=yes`
2. In the new `kitty` terminal, I ran `kitty +kitten ssh a`
3. Using `nnn`'s `preview-tui` yielded the previously reported error

Moreover, I am using `nnn` v4.8 and here is what I get when I run `env | grep -E "(KITTY|NNN)"` on my remote machine:

```
KITTY_PUBLIC_KEY=1:pWEjl`OTAC62kZGD?bZRQRNG=;9Dd#9_U}djB#-Z
KITTY_WINDOW_ID=1
NNN_PLUG=t:preview-tui
NNN_FIFO=/tmp/nnn.fifo
```

and on my local machine:

```
NNN_PLUG=i:imgview;j:autojump;p:preview-tabbed;t:preview-tui;z:fzcd
NNN_FIFO=/tmp/nnn.fifo
NNN_FCOLORS=0B0B04060006060009060B06
KITTY_WINDOW_ID=1
KITTY_PID=13811
KITTY_PUBLIC_KEY=1:pWEjl`OTAC62kZGD?bZRQRNG=;9Dd#9_U}djB#-Z
KITTY_LISTEN_ON=unix:/tmp/kitty-13811
KITTY_INSTALLATION_DIR=/home/alexis/.local/kitty.app/lib/kitty
```

Am I missing anything?
I see that `$KITTY_LISTEN_ON` is not defined on my remote machine, which [makes `nnn` raise an error](https://github.com/jarun/nnn/blob/5d81aeb4776758e1b0e6bc47acdd93dc5e2535b5/plugins/preview-tui#L480). Should I conclude that what I am trying to achieve is not doable?
More generally, shouldn't the ssh connection established with `kitty` try to forward remote actions to the local kitty socket?

**Comment by kovidgoyal** [comment]
kitty will not forward sockets, that would be insanely insecure giving
any program on the remote computer complete control over your local
computer. If that is what you want to do then use

allow_remote_control yes

in kitty.conf

and remote control will work directly over the TTY transparently through
SSH without needing sockets.

**Reply by alexisthual** [reply]
Thank you for tour quick answer!
In this case, I trust the remote machine just as much as I trust mine.

I already set `allow_remote_control yes` in `kitty.conf` on my local machine but it does not seem to have the desired effect. Am I missing anything?

**Comment by kovidgoyal** [comment, marked as answer]
Presumably you have to configure nnn to use the tty rather than a
socket. You need to ask in that project for help.

On Thu, May 04, 2023 at 07:30:52AM +0000, alexis wrote:
> Thank you for tour quick answer!
> In this case, I trust the remote machine just as much as I trust mine.
> 
> I already set ˋallow_remote_control yes` in ˋkitty.conf` on my local machine but it does not seem to have the desired effect. Am I missing anything?
> 
> -- 
> Reply to this email directly or view it on GitHub:
> https://github.com/kovidgoyal/kitty/discussions/6241#discussioncomment-5801774
> You are receiving this because you commented.
> 
> Message ID: ***@***.***>

-- 
_____________________________________

Dr. Kovid Goyal 
https://www.kovidgoyal.net
https://calibre-ebook.com
_____________________________________

**Reply by alexisthual** [reply]
Great, thank you for your help!
For those who are interested in this, I have opened [a discussion there](https://github.com/jarun/nnn/discussions/1637).
